src/contract-details/ContractReceipt.js

```diff
@@ -15,23 +15,30 @@ export default class ContractReceipt extends PureComponent {
 
 	render() {
 		const { contract, showLongcode, onTradeAgainClicked } = this.props;
+		const backendFailure = contract.validation_error && Object.keys(contract).length < 3;
```

Comment:
This is pretty random, failure is indicated when the response contains `contract_id` and `validation_error` only.

I submitted my suggestion in v4 suggestion, and add this change here.